ansible/roles/hadoop/tasks/start-jhs.yml

```diff
@@ -0,0 +1,30 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+- name: Check if JobHistoryServer is running
+  shell: jps | grep "JobHistoryServer" | grep -v grep
```

Comment:
In the past I have had problems with jps because it used to rely on files in /tmp.  Sometimes the OS would clear /tmp and jps would no longer show long running java processes.  Not sure if this still an issue.

Reply:
You can avoid the `grep -v grep` if you make the first grep a pattern: `grep "[J]obHistoryServer"` so that the args to grep don't match the pattern it is looking for.

Also, not sure if it's useful here, but I always run `jps` with `jps -ml`. You probably want to use `-l` here at the very least. Then, you can have a more specific grep pattern, for the fully qualified class name.

Reply:
@ctubbsii you're right! Actually the `grep -v grep` part is not required. I'll remove from this play and also from the `start-zookeeper.yml` in the new commit. Thanks!

@keith-turner, Currently, `jps` command shows the information from the JVM. Even if the /tmp directory was cleared we are still safe as the tool doesn't read from the PID files. I have also tested this manually and it works.

Reply:
@ctubbsii Checking if you are ok with the final changes on this PR? Thanks